packages/calling/src/CallSettings/types.ts

```diff
@@ -90,15 +90,138 @@ export type CallSettingResponse = {
+   *        ringSplashEnabled: true
+   *      },
+   *    },
+   *  message: null
+   * }
+   * ```
+   */
+  getCallWaitingSetting(): Promise<CallSettingResponse>;
+
+  /**
+   * This API is used to fetch the do not disturb(DND) status.
+   *
+   * Example
+   * ```javascript
+   * const dndResponse = await callSettings.getDoNotDisturbSetting();
+   * ```
+   *
+   * The dndResponse object will have callSetting object with the properties as mentioned in `ToggleSetting`.
+   *
+   * Example - ToggleSetting
+   * ```json
+   * {
+   *  statusCode: 200,
+   *    data: {
+   *      callSetting: {
+   *        enabled: true,
+   *        ringSplashEnabled: true
+   *      },
+   *    },
+   *  message: null
+   * }
+   * ```
+   */
+  getDoNotDisturbSetting(): Promise<CallSettingResponse>;
+
+  /**
+   * This API is used to set DND to true or false based on parameter received.
+   *
+   * Example
+   * ```javascript
+   * const dndResponse = await callSettings.setDoNotDisturbSetting(true|false);
+   * ```
+   */
+  setDoNotDisturbSetting(flag: boolean): Promise<CallSettingResponse>;
+
+  /**
+   * This API is used to fetch the call forward setting.
+   *
+   *  Example
+   * ```javascript
+   * const callForwardResponse = await callSettings.getCallForwardSetting();
+   * ```
+   *
+   * The `callForwardResponse` object will have callSetting object with the properties as mentioned in {@link CallForwardSetting}.
+   */
+  getCallForwardSetting(): Promise<CallSettingResponse>;
+
+  /**
+   * This API is used to set the call forward setting.
+   * ```javascript
+   * const callForwardResponse = await callSettings.setCallForwardSetting(callForwardSetting);
+   * ```
+   *
+   * The `callForwardSetting` object will be populated with the properties as mentioned in {@link CallForwardSetting} and passed as a parameter to the API.
+   */
+  setCallForwardSetting(request: CallForwardSetting): Promise<CallSettingResponse>;
+
+  /**
+   * This API is used to fetch the voicemail.
+   *  Example
+   * ```javascript
+   * const voicemailResponse = await callSettings.getVoicemailSetting();
+   * ```
+   *
+   * The `voicemailResponse` object will have callSetting object with the properties as mentioned in {@link VoicemailSetting}.
+   */
+  getVoicemailSetting(): Promise<CallSettingResponse>;
+
+  /**
+   * This API is used to set voicemail.
+   *   Example
+   * ```javascript
+   * const voicemailResponse = await callSettings.setVoicemailSetting();
+   * ```
+   *
+   * The `voicemailSetting` object will be populated with the properties as mentioned in {@link VoicemailSetting} and passed as a parameter to the API.
+   */
+  setVoicemailSetting(request: VoicemailSetting): Promise<CallSettingResponse>;
+
+  /**
+   * This API is used to fetch the call forward settings including the Voicemail.
+   * @param directoryNumber - Directory number for which the call forward always setting is to be fetched.
+   *
+   * This parameter is only required for CCUC backend.
```

Comment:
Would the end customer know what a CCUC backend is?

Reply:
Added the full-form as well as the link